docs/navigation-context.md

```diff
@@ -0,0 +1,43 @@
+---
+id: navigation-context
+title: NavigationContext
+sidebar_label: NavigationContext
+---
+
+`NavigationContext` provides the `navigation` object (similar to the [navigation](navigation-prop.html) prop). In fact, [withNavigation](with-navigation.html) uses this context to inject the `navigation` prop to your wrapped component. The [hook counterpart](https://github.com/react-navigation/react-navigation-hooks#usenavigation) is essentially an `useContext` with this context as well.
+
+Most of the time, you will not likely yo use this Context directly, as the provided `withNavigation` and [hooks](https://github.com/react-navigation/react-navigation-hooks) already cover all the possible use cases.
```

Comment:
```suggestion
Most of the time, you will use not use `NavigationContext` directly, as the provided `withNavigation` and [hooks](https://github.com/react-navigation/react-navigation-hooks) already cover most use cases. But just in case you have something else in mind, `NavigationContext` is available for you to use.
```